packages/core/src/amazonq/webview/ui/apps/cwChatConnector.ts

```diff
@@ -183,17 +192,29 @@ export class Connector {
         })
     }
 
-    requestGenerativeAIAnswer = (tabID: string, payload: ChatPayload): Promise<any> =>
-        new Promise((resolve, reject) => {
+    requestGenerativeAIAnswer = (tabID: string, messageId: string, payload: ChatPayload): Promise<any> => {
+        /**
+         * When a user presses "enter" send an event that indicates
+         * we should start tracking the round trip time for this message
+         **/
+        this.sendMessageToExtension({
+            command: 'start-chat-message-telemetry',
+            trigger: 'onChatPrompt',
+            tabID,
+            traceId: messageId,
+            tabType: 'cwc',
+            startTime: Date.now(),
```

Comment:
Use `globals.clock.Date.now()` since if we eventually want to stub the time with `installFakeClock()` it will not work with the way things currently are

Reply:
🤦